src/lib/zkapp_command_builder/zkapp_command_builder.ml

```diff
@@ -59,25 +59,30 @@ let mk_zkapp_command ?memo ~fee ~fee_payer_pk ~fee_payer_nonce account_updates :
     Option.value_map memo ~default:Signed_command_memo.dummy
       ~f:Signed_command_memo.create_from_string_exn
   in
-  { fee_payer
-  ; memo
-  ; account_updates =
-      account_updates
-      |> Zkapp_command.Call_forest.map
-           ~f:(fun (p : Account_update.Body.Simple.t) : Account_update.t ->
-             let authorization =
-               match p.authorization_kind with
-               | None_given ->
-                   Control.Poly.None_given
-               | Proof _ ->
-                   Control.Poly.Proof
-                     (Lazy.force Mina_base.Proof.blockchain_dummy)
-               | Signature ->
-                   Control.Poly.Signature Signature.dummy
-             in
-             { body = Account_update.Body.of_simple p; authorization } )
-      |> Zkapp_command.Call_forest.accumulate_hashes_predicated
-  }
+  Zkapp_command.map_proofs
+    ~f:(const @@ Lazy.force Proof.For_tests.blockchain_dummy_tag)
+    { Zkapp_command.Poly.fee_payer
+    ; memo
+    ; account_updates =
+        Zkapp_command.Call_forest.map account_updates
+          ~f:(fun (p : Account_update.Body.Simple.t) ->
+            let authorization =
+              match p.authorization_kind with
+              | None_given ->
+                  Control.Poly.None_given
+              | Proof _ ->
+                  Control.Poly.Proof
+                    (Lazy.force Mina_base.Proof.blockchain_dummy)
+              | Signature ->
+                  Control.Poly.Signature Signature.dummy
+            in
+            { Account_update.Poly.body = Account_update.Body.of_simple p
+            ; authorization
+            } )
+        |> Zkapp_command.Call_forest.accumulate_hashes_predicated
+    }
+
+let proof_cache_db = Proof_cache_tag.For_tests.create_db ()
```

Comment:
I'm wondering why we have this top-level value. It will always be initialized. We should ensure that the library is unused in executables.